### PR TITLE
Single definition fetching only works when tags are present

### PR DESCRIPTION
**Pull request by burmanm**
### https://issues.jboss.org/browse/HWKMETRICS-110

Do not merge yet (tests are failing on my machine for all tests, so I can't verify this yet)

* Change findMetrics to return Observable<Metric instead of Optional (doesn't make much sense with RxJava to use Optional)
* Add some tests and try to fix the fetching single definition

Do not merge yet, @jsanda could you comment if the approach to reading from metrics_idx is the right way?

### CODE REVIEW

**Comment by jsanda** [comment]
I agree that using Optional does not make much sense. +1 on that change. With these changes, can we get rid of the ```DataAccessImpl.findMetric``` query/field?

File: api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/AvailabilityHandler.java [hunk 1]
```diff
@@ -123,8 +123,10 @@ public void getAvailabilityMetric(@Suspended final AsyncResponse asyncResponse,
             @HeaderParam("tenantId") String tenantId, @PathParam("id") String id) {
 
         metricsService.findMetric(tenantId, MetricType.AVAILABILITY, new MetricId(id))
+                .map(m -> Response.ok(m).build())
+                .defaultIfEmpty(Response.noContent().build())
```
Comment (burmanm) [inline api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/AvailabilityHandler.java]: This defaultIfEmpty() is triggered even when there's rows to map(). Will have to wait until I'm back.. removing that one will return correctly the definition.

**Comment by tsegismont** [comment]
@burmanm where do you stand with this one?

**Comment by burmanm** [comment]
Replaced by PR #278

**Pull request closed without merging**